### ISSUE TITLE
Slight routing issue detected during 1.4.3 -> 1.5.1 upgrade process Kong + Cassandra

### ISSUE THREAD

**Issue by jeremyjpj0916** [Kong/kong#5985]
# Summary 

Oddly when going from 1.4.3 to 1.5.1, 2 of our 4 prod Kong clusters faced a weird breaking behavior on 1 proxy in each. The behavior is best explained by reviewing our splunk logs for the good before behavior (1.4.3) and bad after behavior (1.5.1):

Before: 1.4.3

```
{ 
   BackendLatency: 39
   CID: 2f4e4008-79df-48fe-9ae5-f3111603ae7e#41729
   ClientIP: 10.xxx.xxx.xx
   Consumer: consumer1
   GatewayHost: gateway-company.com
   GatewayPort: 443
   HTTPMethod: POST
   HTTPStatus: 200
   RequestSize: 764
   ResponseSize: 633
   RoutingURL: my-cool-api.com:443/api/pathing/v1.0/logs -> Backend URL + wildcard pathing appended
   ServiceName: api.prod
   TotalLatency: 53
   URI: /api/pathing/v1.0/logs -> Proxy URI seen called on gateway
}
```

After: 1.5.1 

```
{
   BackendLatency: 10
   CID: cba2ef8b-3ebc-4dec-aed6-09f3de0b7f8f#31566
   ClientIP: 10.xxx.xxx.xx
   Consumer: consumer1
   GatewayHost: gateway-company.com
   GatewayPort: 443
   HTTPMethod: POST
   HTTPStatus: 404
   RequestSize: 2151
   ResponseSize: 2325
   RoutingURL: my-cool-api.com:443/api/pathing/v1/.0/logs
   ServiceName: api.prod
   TotalLatency: 27
   URI: /api/pathing/v1.0/logs
}
```

All our proxies leverage wild card routing, and basically the proxy route was defined up to the ```/v1``` and the backend URL on the service ended at ```/v1```, then customers were appending an extra ```.0/logs``` to generate traffic to what was expected ```/v1.0/logs``` , seems post migration 2 proxies out of our 880+ proxies between our various prod environments. 

For context, the route and service resource info(that we left unchanged between versions):

Service:
```
{
  "host": "my-cool-api.com",
  "created_at": 1578040896,
  "connect_timeout": 2000,
  "id": "3afda9be-13b9-4b9b-8221-ba2599f7673f",
  "protocol": "https",
  "name": "api.prod",
  "read_timeout": 9000,
  "port": 443,
  "path": "/api/pathing/v1",
  "updated_at": 1591168467,
  "retries": 0,
  "write_timeout": 9000,
  "tags": null,
  "client_certificate": null,
  "extras": {}
}
```

Route:

```
{
  "strip_path": true,
  "path_handling": "v1",
  "paths": [
    "/api/pathing/v1"
  ],
  "destinations": null,
  "headers": null,
  "protocols": [
    "http",
    "https"
  ],
  "methods": null,
  "snis": null,
  "service": {
    "id": "3afda9be-13b9-4b9b-8221-ba2599f7673f"
  },
  "name": null,
  "updated_at": 1591168513,
  "preserve_host": false,
  "regex_priority": 0,
  "hosts": null,
  "sources": null,
  "id": "0ced64ce-0a34-40bc-81ae-01bca739a01a",
  "https_redirect_status_code": 426,
  "tags": null,
  "created_at": 1578040896
}
```

Related likely:
https://github.com/Kong/kong/pull/3780

Now to fix the issue on the 4am warroom I attended, I just popped open the Konga UI, re-submitted the service resource against the admin api(assuming konga UI sends patch there) with just the same info in the fields as when the resource loaded up untouched. Then I also opened the route resource up and also just submitted what was up. Seems like that cache clearing of sorts fixed the issue. What perplexes me is why only 2 proxies saw this issue and not more if it was just 100% a pathing error because many of our wild card proxies would likely have been messed up I imagine throwing an extra ```/``` on the backend URL like that. 

To hunt down the 2 proxies showing this behavior (1 was reported by customer), I used a query like this in splunk:
```
index=cba_gateway RoutingURL=*/.0*
```
To find backend pathings with a / between what would likely be api versioning taxonomy standard. 1.0 , 2.0 etc. 

Figured I would let yah know incase you see it as well somehow with some enterprise users as upgrades occur.

### Steps To Reproduce

1. See the before and after of logs and my explanation for insight.

**Comment by jeremyjpj0916** [comment]
Ah this table is a real info dump madness hah. So from 1.4.3 we in theory were already on v1 right(by nature since 1.x, we never touched router versions). routes after the migration that even work now still in Konga read as v1, so in theory everything was v1 and should have stayed v1 from the migration jump. All it took was opening the Konga UI and resubmitting the data, wish I had reviewed the json contents of those resources right during the broken impact to see if it had been v0 or v1 but that data is lost now, all I can say is that PATCHING the resource by submitting the same data Konga had stored on the 2 broken routes fixed it. And all routes are made the same way in our Kong so was weird for 2 of them to behave like they did. Let me try changing from v1 back to v0 in a lab environment and see if the breaking behavior I witnessed on the two prod proxies end up reflecting the same behavior on v0 in our lab env.
